src/api/Shrooms.Infrastructure/Email/ISmtpService.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections.Generic;
+using System.Net.Mail;
+using System.Threading.Tasks;
+
+namespace Shrooms.Infrastructure.Email
+{
+    /// <summary>
+    /// Service that wraps SMTP client and SMTP mail settings.
+    /// </summary>
+    public interface ISmtpService
+    {
+        /// <summary>
+        /// Sends mail messages via SMTP asynchronously.
+        /// </summary>
+        /// <param name="messages">Message collection for sending.</param>
+        /// <returns>A <see cref="Task"/> that represents asynchronous operation.</returns>
+        Task SendAsync(IEnumerable<MailMessage> messages);
+
+        /// <summary>
+        /// Determines if SMTP configuration is present and valid.
+        /// </summary>
+        bool HasSmtpServerConfigured();
```

Comment:
```suggestion
        bool IsSmtpServerConfigured();
```

I would say it would be more natural name. Otherwise it's not clear who "has". Even in comment you wrote "... configuration IS present". (I know this name was there before, but since you are refactoring it anyway.)